src/vsr/superblock.zig

```diff
@@ -1024,11 +1042,11 @@ pub fn SuperBlockType(comptime Storage: type) type {
 
         fn write_trailer_callback(write: *Storage.Write) void {
             const context = @fieldParentPtr(Context, "write", write);
-            const trailer_ = context.write_trailer.?;
+            const trailer = context.trailer.?;
 
-            context.write_trailer = null;
+            context.trailer = null;
```

Comment:
Alternatively, instead of going `null -> .manifest -> null -> .free_set -> null -> .client_session -> null`, we could go `null -> .manifest -> .free_set -> .client_session -> null`, as that would narrow down the state transition somewhat.